src/lightning_app/cli/cmd_clusters.py

```diff
@@ -0,0 +1,213 @@
+            )
+            click.confirm("Do you want to continue?", abort=True)
+
+        self.api_client.cluster_service_delete_cluster(id=cluster_id, force=force)
+        click.echo("Cluster deletion triggered successfully")
+
+        if wait:
+            _wait_for_cluster_state(self.api_client, cluster_id, V1ClusterState.DELETED)
+
+
+class ClusterList(Formatable):
+    def __init__(self, clusters: [Externalv1Cluster]):
+        self.clusters = clusters
+
+    def as_json(self) -> str:
+        return json.dumps(self.clusters)
+
+    def as_table(self) -> Table:
+        table = Table("id", "name", "type", "status", "created", show_header=True, header_style="bold green")
+        phases = {
+            V1ClusterState.QUEUED: Text("queued", style="bold yellow"),
+            V1ClusterState.PENDING: Text("pending", style="bold yellow"),
+            V1ClusterState.RUNNING: Text("running", style="bold green"),
+            V1ClusterState.FAILED: Text("failed", style="bold red"),
+            V1ClusterState.DELETED: Text("deleted", style="bold red"),
+        }
+
+        cluster_type_lookup = {
+            V1ClusterType.BYOC: Text("byoc", style="bold yellow"),
+            V1ClusterType.GLOBAL: Text("lightning-cloud", style="bold green"),
+        }
+        for cluster in self.clusters:
+            cluster: Externalv1Cluster
+            status = phases[cluster.status.phase]
+            if cluster.spec.desired_state == V1ClusterState.DELETED and cluster.status.phase != V1ClusterState.DELETED:
+                status = Text("terminating", style="bold red")
+
+            # this guard is necessary only until 0.3.93 releases which includes the `created_at`
+            # field to the external API
+            created_at = datetime.now()
+            if hasattr(cluster, "created_at"):
+                created_at = cluster.created_at
+
+            table.add_row(
+                cluster.id,
+                cluster.name,
+                cluster_type_lookup.get(cluster.spec.cluster_type, Text("unknown", style="red")),
+                status,
+                arrow.get(created_at).humanize() if created_at else "",
+            )
+        return table
+
+
+def _wait_for_cluster_state(
+    api_client,
+    cluster_id: str,
+    target_state: V1ClusterState,
+    max_wait_time=MAX_CLUSTER_WAIT_TIME,
+    check_timeout=CLUSTER_STATE_CHECKING_TIMEOUT,
+):
+    start = time.time()
+    elapsed = 0
+    while elapsed < max_wait_time:
+        cluster_resp = api_client.cluster_service_list_clusters()
+        new_cluster = None
+        for clust in cluster_resp.clusters:
+            if clust.id == cluster_id:
+                new_cluster = clust
+                break
+        if new_cluster is not None:
+            if new_cluster.status.phase == target_state:
+                break
+            elif new_cluster.status.phase == V1ClusterState.FAILED:
+                raise click.ClickException(f"cluster {cluster_id} is in failed state")
```

Comment:
What exactly is a "failed state"?

Please change to:

Cluster {cluster_id} is in a failed state.

Reply:
failed state means the provisioning did not complete - this usually happens when the cloud provider being targetted has a problem - like quota limits, or outages/ validation issues.

At this point we can't provide the user with a more detailed reason unfortunately.